site/en/docs/extensions/mv3/sandboxingEval/index.md

```diff
@@ -64,42 +64,94 @@ named "sandbox.html". The manifest entry looks like this:
 ### Load the sandboxed file {: #load_file }
 
 In order to do something interesting with the sandboxed file, we need to load it in a context where
-it can be addressed by the extension's code. Here, [sandbox.html][8] has been loaded into the
-extension's [Event Page][9] ([eventpage.html][10]) via an `iframe`. [eventpage.js][11] contains code
-that sends a message into the sandbox whenever the browser action is clicked by finding the `iframe`
+it can be addressed by the extension's code. Here, sandbox.html has been loaded into
+an extension page via an `iframe`. The page's javascript file contains code that sends a message 
+into the sandbox whenever the browser action is clicked by finding the `iframe`
 on the page, and executing the `postMessage` method on its `contentWindow`. The message is an object
-containing two properties: `context` and `command`. We'll dive into both in a moment.
+containing two properties: `templateName`, and `command`. We'll dive into both in a moment.
+
+{% Label %}service-worker.js:{% endLabel %}
 
 ```js
-chrome.browserAction.onClicked.addListener(function() {
- var iframe = document.getElementById('theFrame');
- var message = {
-   command: 'render',
-   context: {thing: 'world'}
- };
- iframe.contentWindow.postMessage(message, '*');
+chrome.runtime.onInstalled.addListener(() => {
+  chrome.tabs.create({
+    url: 'mainpage.html'
+  });
+  console.log('Opened a tab with a sandboxed page!');
 });
 ```
 
-<div class="aside aside--note">For general information about the <code>postMessage</code> API, take a look at the <a href="https://developer.mozilla.org/en/DOM/window.postMessage"><code>postMessage</code> documentation on MDN </a>. It's quite complete and worth reading. In particular, note that data can only be passed back and forth if it's serializable. Functions, for instance, are not.</div>
+{% Label %}extension-page.js:{% endLabel %}
+
+```js
+let counter = 0;
```

Comment:
curious, why do we need the counter?

Reply:
@sebastianbenz For both of these code concerns, Oliver advised I keep the example code blocks true to the exact code referenced in the sample repository. (I had previously simplified it a bit for the documentation). 
Should I make these edits both here and in a new PR to the sample itself?

Reply:
sorry, forgot to remove the counter comment, saw later that this is displayed.

But, yes, please make both changes for the onInstalled -> action on sample and doc. Thanks!